.github/workflows/docker.yml

```diff
@@ -191,6 +191,9 @@ jobs:
               echo "DOCKER_PUSH_REPOSITORY=$(echo ${{ inputs.docker_push_repository }} | tr "[:upper:]" "[:lower:]")" >> $GITHUB_ENV
               echo "DOCKER_CONFIG_FILE=$HOME/.docker/config.json" >> $GITHUB_ENV
             fi
+
+      - name: Determine Docker tags to use
+        run: |
             # This line needs to be run before the step "Merge CI fixes from sagemath/sage".
```

Comment:
This fix seems to indicate that the previous step (logging to ghcr.io) is not really necessary (used) for subsequent steps. So perhaps the step is here to provide the environment variables to the calling workflow.

Reply:
The environment variables are set for the following steps in this job, specifically for invoking `tox`.

Reply:
I meant 

(A): logging into docker and setting the environment variables `DOCKER_PUSH_REPOSITORY` and `DOCKER_CONFIG_FILE`

 in the previous step to the "Determine Docker tags to use" step.

If the idea is to run "Determine Docker tags to use" step even when `inputs.docker_push_repository` is empty, then it means that (A) is not necessary for the steps following and including the "Determine Docker tags to use" step.

So I thought (A) could be moved out to the calling workflow and don't have to be in the docker workflow. I am not suggesting we should do it. I just want to know if I understand the situation correctly. Am I right?

Reply:
> If the idea is to run "Determine Docker tags to use" step even when `inputs.docker_push_repository` is empty, then it means that (A) is not necessary for the steps following and including the "Determine Docker tags to use" step.

That's right, the two parts now separated do not interact with each other. Each part sets some environment variables for the following steps.

> So I thought (A) could be moved out to the calling workflow and don't have to be in the docker workflow. I am not suggesting we should do it. I just want to know if I understand the situation correctly. Am I right?

No; environment variables set by appending to the file `$GITHUB_ENV` only take effect for the next steps of a job. But a reusable workflow is run as a job, not a step of a job. See https://docs.github.com/en/actions/using-workflows/reusing-workflows#limitations

Reply:
OK. Thanks.